libraries/Parsers/Microsoft.Bot.Builder.Parsers.LU/Parser/Visitor.cs

```diff
@@ -0,0 +1,224 @@
+        {
+            char[] invalidCharsInIntentOrEntityName = { '<', '>', '*', '%', '&', ':', '\\', '$' };
+            foreach (var item in tokenizedUterance)
+            {
+                if (item is AuxEntity auxEntity)
+                {
+                    var entityName = auxEntity.EntityName.Trim();
+                    if (!string.IsNullOrEmpty(entityName) && entityName.IndexOfAny(invalidCharsInIntentOrEntityName) >= 0)
+                    {
+                        errorMessages.Add($"Invalid utterance line, entity name {entityName} cannot contain any of the following characters: [<, >, *, %, &, :, \\, $]");
+                        continue;
+                    }
+
+                    if (auxEntity.EntityValue == null)
+                    {
+                        // we have a pattern.any entity
+                        var patternStr = !string.IsNullOrEmpty(auxEntity.Role) ? $"{{{auxEntity.EntityName}:{auxEntity.Role}}}" : $"{{{auxEntity.EntityName}}}";
+                        srcUtterance.Append(patternStr);
+                        entities.Add(
+                            new EntityElement
+                            {
+                                Type = TypeEnum.PatternAnyEntities,
+                                Entity = entityName,
+                                Role = auxEntity.Role.Trim()
+                            });
+                    }
+                    else
+                    {
+                        // we have a new entity
+                        var newEntity = new EntityElement
+                        {
+                            Type = TypeEnum.Entities,
+                            Entity = entityName,
+                            Role = auxEntity.Role.Trim(),
+                            StartPos = srcUtterance.Length,
+                            EndPos = null
+                        };
+                        if (auxEntity.EntityValue == null)
+                        {
+                            errorMessages.Add($"Composite entity {auxEntity.Parent.EntityName} includes pattern.any entity {auxEntity.EntityName}.\r\n\tComposites cannot include pattern.any entity as a child.");
+                        }
+                        else
+                        {
+                            srcUtterance = new StringBuilder(RecursivelyResolveTokenizedUtterance(auxEntity.EntityValue, entities, errorMessages, srcUtterance).ToString().TrimStart());
+                            newEntity.EndPos = srcUtterance.Length - 1;
+                            entities.Add(newEntity);
+                        }
+                    }
+                }
+                else
+                {
+                    string unicodeCharItem = (string)item;
+                    srcUtterance.Append(unicodeCharItem);
+                }
+            }
+
+            return srcUtterance;
+        }
+
+        private static List<object> TokenizeUtterance(string expression)
```

Comment:
Is this `List<object>` because of the covariance / contravariance problem we discussed? Do we know nothing about the types of the objects we have in the lists? i.e. no common parent class, etc? Is there  aplace where we cast these objects to specific types? Where is that?